Fix/2.0 style scope

tomivirkki opened the pull request:
Closes #538

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-grid/568)
<!-- Reviewable:end -->

Saulis commented:
Review status: 0 of 11 files reviewed at latest revision, 2 unresolved discussions.

---

*[vaadin-grid-cell-click-behavior.html, line 20 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/568#-KYrC4aZIzVZo1tRPtQN:-KYrC4aZIzVZo1tRPtQO:b-piu46v) ([raw file](https://github.com/vaadin/vaadin-grid/blob/164e72d3cdfe960da32c6b3a0704d21ba2b3c6e1/vaadin-grid-cell-click-behavior.html#L20)):*
> ```HTML
>     _onClick: function(e) {
>       // Prevent item action if cell itself is not focused.
>       if (!this._isFocusable(Polymer.dom(e).rootTarget)) {
> ```

This will always return `false` for elements like `<paper-checkbox>` which don't have focusable elements inside their local dom. We might need to change the whole `isFocusable` logic to check if the `document.activeElement` is on the `event.path` between the cell and the `rootTarget`. @platosha has some ideas about this.

---

*[vaadin-grid-table-cell.html, line 139 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/568#-KYrD5hvab6NMchZQjKB:-KYrD5hvab6NMchZQjKC:bkaddnt) ([raw file](https://github.com/vaadin/vaadin-grid/blob/164e72d3cdfe960da32c6b3a0704d21ba2b3c6e1/vaadin-grid-table-cell.html#L139)):*
> ```HTML
>           this.debounce('remove-content', function() {
>             if (!this.isAttached && cellContent.parentNode) {
>               Polymer.dom(this.target).removeChild(cellContent);
> ```

Just as a comment: a cell is not expected to be re-attached after detaching, so there is no need to remove the `insertionPoint` manually. Anyways, the re-appending wouldn't do anything since it's already in the correct place.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/568)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-grid-cell-click-behavior.html, line 20 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/568#-KYrC4aZIzVZo1tRPtQN:-KYrMgXFb29DTVxfsPp_:b-1sz3v7) ([raw file](https://github.com/vaadin/vaadin-grid/blob/164e72d3cdfe960da32c6b3a0704d21ba2b3c6e1/vaadin-grid-cell-click-behavior.html#L20)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

This will always return `false` for elements like `<paper-checkbox>` which don't have focusable elements inside their local dom. We might need to change the whole `isFocusable` logic to check if the `document.activeElement` is on the `event.path` between the cell and the `rootTarget`. @platosha has some ideas about this.
</blockquote></details>

Reverted the change. The original problem was that .localTarget is the `<content>` element on shady DOM. Changed the fix so that when necessary, 'target' is reassigned with cell-content wrapper from the content element's distributed nodes.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/568)*
<!-- Sent from Reviewable.io -->

Saulis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> :shipit: !

---

Reviewed 10 of 11 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-grid-cell-click-behavior.html, line 20 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/568#-KYrC4aZIzVZo1tRPtQN:-KYrUfn385M1kmyJyQFC:b-faa0qd) ([raw file](https://github.com/vaadin/vaadin-grid/blob/164e72d3cdfe960da32c6b3a0704d21ba2b3c6e1/vaadin-grid-cell-click-behavior.html#L20)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Reverted the change. The original problem was that .localTarget is the `<content>` element on shady DOM. Changed the fix so that when necessary, 'target' is reassigned with cell-content wrapper from the content element's distributed nodes.
</blockquote></details>

I think this is fine for now – we can improve on it later if someone bumps into concrete issues with this approach.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/568#-:-KYrVlcTi0mcLJ3unLmq:bj7fgrt)*
<!-- Sent from Reviewable.io -->
